app/src/main/java/com/ataulm/artcollector/Navigation.kt

```diff
@@ -3,20 +3,33 @@ package com.ataulm.artcollector
 import android.content.Intent
 import android.net.Uri
 
-enum class Navigation(private val semiQualifiedActivityName: String) {
+private const val SCHEME = "https"
+private const val AUTHORITY = "art-collector.ataulm.com"
+private const val ARTIST_GALLERY = "${BuildConfig.APPLICATION_ID}.artist.ui.ArtistActivity"
+private const val PAINTING = "${BuildConfig.APPLICATION_ID}.painting.ui.PaintingActivity"
 
-    GALLERY("gallery.ui.GalleryActivity"),
-    PAINTING("painting.ui.PaintingActivity");
+fun artistGalleryIntent(artistId: String): Intent {
+    val uri = Uri.Builder()
+            .scheme(SCHEME)
+            .authority(AUTHORITY)
+            .path(artistId)
+            .build()
 
-    fun viewIntent(artistId: String, paintingId: String): Intent {
-        val uri = Uri.Builder()
-                .scheme("https://")
-                .authority("art-collector.ataulm.com")
-                .path("$artistId/$paintingId")
-                .build()
+    return intent(uri, ARTIST_GALLERY)
+}
+
+fun paintingIntent(artistId: String, paintingId: String): Intent {
+    val uri = Uri.Builder()
+            .scheme(SCHEME)
+            .authority(AUTHORITY)
+            .path("$artistId/$paintingId")
+            .build()
+
+    return intent(uri, PAINTING)
+}
 
-        return Intent(Intent.ACTION_VIEW)
-                .setClassName(BuildConfig.APPLICATION_ID, "${BuildConfig.APPLICATION_ID}.$semiQualifiedActivityName")
-                .setData(uri)
-    }
+private fun intent(uri: Uri, componentName: String): Intent {
+    return Intent(Intent.ACTION_VIEW)
+            .setClassName(BuildConfig.APPLICATION_ID, componentName)
```

Comment:
we should just be able to ditch this line when we add the correct intent filters in the manifest 🤔